pentasharp/Controllers/BusSchedule.cs

```diff
@@ -1,21 +1,336 @@
+        {
+            var route = _context.BusRoutes.Find(id);
+            if (route == null)
+                return NotFound(new { success = false, message = "Route not found." });
+
+            if (_context.BusRoutes.Any(r => r.RouteId != id && r.FromLocation == model.FromLocation && r.ToLocation == model.ToLocation))
+                return BadRequest(new { success = false, message = "A route with the same details already exists." });
+
+            route.FromLocation = model.FromLocation;
+            route.ToLocation = model.ToLocation;
+            route.EstimatedDuration = new TimeSpan(hours, minutes, 0);
+
+            _context.SaveChanges();
+            return Ok(new { success = true, message = "Route updated successfully." });
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpDelete("DeleteRoute/{id}")]
+        public IActionResult DeleteRoute(int id)
+        {
+            var route = _context.BusRoutes.Find(id);
+            if (route == null)
+                return NotFound(new { success = false, message = "Route not found." });
+
+            _context.BusRoutes.Remove(route);
+            _context.SaveChanges();
+
+            return Ok(new { success = true, message = "Route deleted successfully." });
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpPost("AddSchedule")]
+        public IActionResult AddSchedule([FromBody] AddScheduleViewModel model)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(new { success = false, message = "Invalid data provided." });
+
+            var route = _context.BusRoutes.FirstOrDefault(r => r.RouteId == model.RouteId);
+            if (route == null)
+                return BadRequest(new { success = false, message = "Route not found." });
+
+            var bus = _context.Buses.FirstOrDefault(b => b.BusId == model.BusId);
+            if (bus == null)
+                return BadRequest(new { success = false, message = "Bus not found." });
+
+            if (_context.BusSchedules.Any(s => s.BusId == model.BusId && s.DepartureTime == model.DepartureTime))
+                return BadRequest(new { success = false, message = "A schedule for the same bus at the same time already exists." });
+
+            model.ArrivalTime = model.DepartureTime.Add(route.EstimatedDuration);
+            model.AvailableSeats = bus.Capacity;
+
+            var schedule = _mapper.Map<BusSchedule>(model);
+            schedule.Status = BusScheduleStatus.Scheduled;
+
+            _context.BusSchedules.Add(schedule);
+            _context.SaveChanges();
+
+            return Ok(new { success = true, message = "Schedule added successfully." });
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpGet("GetSchedules")]
+        public IActionResult GetSchedules()
+        {
+            var schedules = _context.BusSchedules.Include(s => s.Route).Include(s => s.Bus).ToList();
```

Comment:
use LINQ, the reason behind it is that using .Include will fetch all fields of Route and Bus entities, even though we dont need all of them, also after loading all those data (including unnecessary ones) they will all be applied in memory rather than being optimized with database query.